public/src/main/java/com/revenuecat/purchases/errors.kt

```diff
@@ -0,0 +1,53 @@
+package com.revenuecat.purchases
+
+/**
+ * This class represents an error
+ * @param code Error code
+ * @param underlyingErrorMessage Optional error message with a more detailed explanation of the
+ * error that originated this.
+ */
+class PurchasesError(
+    val code: PurchasesErrorCode,
+    val underlyingErrorMessage: String? = null
+) {
+    // Message explaining the error
+    val message: String = code.description
+
+    // TODO: comment this out? what is this?
+//    init {
+//        errorLog("${code.description}${underlyingErrorMessage?.let { " | $it" }}")
+//    }
```

Comment:
So besides the fact that `errorLog` can't be accessed from this module, I don't think this log should happen here. This should live in the `toPurchasesError()` functions that build these errros